File: mobile/src/main/java/org/openhab/habdroid/ui/activity/ContentControllerOnePane.java
/*
 * Copyright (c) 2018, openHAB.org and others.
 *
 *   All rights reserved. This program and the accompanying materials
 *   are made available under the terms of the Eclipse Public License v1.0
 *   which accompanies this distribution, and is available at
 *   http://www.eclipse.org/legal/epl-v10.html
 */

package org.openhab.habdroid.ui.activity;

import android.view.ViewStub;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import org.openhab.habdroid.R;
import org.openhab.habdroid.ui.MainActivity;
import org.openhab.habdroid.ui.WidgetListFragment;

@SuppressWarnings("unused") // instantiated via reflection
public class ContentControllerOnePane extends ContentController {
    public ContentControllerOnePane(MainActivity activity) {
        super(activity);
    }

    @Override
    protected void executeStateUpdate(FragmentUpdateReason reason, boolean allowStateLoss) {
        Fragment currentFragment = mFm.findFragmentById(R.id.content);
        Fragment fragment = getOverridingFragment();
        if (fragment == null && !mPageStack.isEmpty()) {
            fragment = mPageStack.peek().second;
        }
        if (fragment == null) {
            fragment = mSitemapFragment;
        }

        FragmentTransaction ft = mFm.beginTransaction()
                .setCustomAnimations(determineEnterAnim(reason), determineExitAnim(reason))
                .replace(R.id.content, fragment != null ? fragment : mDefaultProgressFragment);
        if (allowStateLoss) {
            ft.commitAllowingStateLoss();
        } else {
            ft.commit();
        }
    }

    @Override
    protected WidgetListFragment getFragmentForTitle() {
        return mPageStack.empty() ? mSitemapFragment : mPageStack.peek().second;
    }

    @Override
    public void inflateViews(ViewStub stub) {
        stub.setLayoutResource(R.layout.content_onepane);
        stub.inflate();
    }
}

File: mobile/src/main/java/org/openhab/habdroid/ui/activity/ContentControllerOnePane.kt
/*
 * Copyright (c) 2010-2023 Contributors to the openHAB project
 *
 * See the NOTICE file(s) distributed with this work for additional
 * information.
 *
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License 2.0 which is available at
 * http://www.eclipse.org/legal/epl-2.0
 *
 * SPDX-License-Identifier: EPL-2.0
 */

package org.openhab.habdroid.ui.activity

import android.view.View
import android.view.ViewStub
import androidx.annotation.Keep
import androidx.fragment.app.commit
import org.openhab.habdroid.R
import org.openhab.habdroid.ui.MainActivity

// instantiated via reflection
@Suppress("UNUSED")
@Keep
class ContentControllerOnePane(activity: MainActivity) : ContentController(activity) {
    override val fragmentForTitle get() = if (pageStack.empty()) sitemapFragment else pageStack.peek().second
    override val fragmentForAppBarScroll get() = fragmentForTitle

    override fun executeStateUpdate(reason: FragmentUpdateReason) {
        val fragment = when {
            overridingFragment != null -> overridingFragment
            !pageStack.isEmpty() -> pageStack.peek().second
            else -> sitemapFragment
        }

        fm.commit(!fm.isStateSaved) {
            setCustomAnimations(determineEnterAnim(reason), determineExitAnim(reason))
            replace(R.id.content, fragment ?: defaultProgressFragment)
        }
    }

    override fun inflateContentView(stub: ViewStub): View {
        stub.layoutResource = R.layout.content_onepane
        return stub.inflate()
    }
}
